cnxepub/formatters.py

```diff
@@ -319,9 +318,63 @@ def _fix_namespaces(html):
                           method='html', encoding='utf-8')
 
 
+def exercise_callback_factory(match, url_template):
+    """Create a callback function to replace an exercise by fetching from
+    a server."""
+    import requests
+
+    def _replace_exercises(elem):
+        item_code = elem.get('href')[len(match):]
+        url = url_template.format(itemCode=item_code)
+        res = requests.get(url)
+        if res:
```

Comment:
Could/should this error instead of silently failing? Otherwise, it seems like people would incorrectly think that a book is successfully generated.